src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -56,50 +48,59 @@ public LanguageServerProjectSystem(
-            var buildHost = await buildHostProcessManager.GetBuildHostAsync(BuildHostProcessKind.NetCore, CancellationToken.None);
+        _logger.LogInformation(string.Format(LanguageServerResources.Loading_0, solutionFilePath));
+        ProjectFactory.SolutionPath = solutionFilePath;
 
-            // If we don't have a .NET Core SDK on this machine at all, try .NET Framework
-            if (!await buildHost.HasUsableMSBuildAsync(solutionFilePath, CancellationToken.None))
-            {
-                var kind = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? BuildHostProcessKind.NetFramework : BuildHostProcessKind.Mono;
-                buildHost = await buildHostProcessManager.GetBuildHostAsync(kind, CancellationToken.None);
-            }
+        // We'll load solutions out-of-proc, since it's possible we might be running on a runtime that doesn't have a matching SDK installed,
+        // and we don't want any MSBuild registration to set environment variables in our process that might impact child processes.
+        await using var buildHostProcessManager = new BuildHostProcessManager(globalMSBuildProperties: AdditionalProperties, loggerFactory: LoggerFactory);
+        var buildHost = await buildHostProcessManager.GetBuildHostAsync(BuildHostProcessKind.NetCore, CancellationToken.None);
 
-            foreach (var project in await buildHost.GetProjectsInSolutionAsync(solutionFilePath, CancellationToken.None))
-            {
-                ProjectsToLoadAndReload.AddWork(new ProjectToLoad(project.ProjectPath, project.ProjectGuid, ReportTelemetry: true));
-            }
+        // If we don't have a .NET Core SDK on this machine at all, try .NET Framework
+        if (!await buildHost.HasUsableMSBuildAsync(solutionFilePath, CancellationToken.None))
+        {
+            var kind = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? BuildHostProcessKind.NetFramework : BuildHostProcessKind.Mono;
+            buildHost = await buildHostProcessManager.GetBuildHostAsync(kind, CancellationToken.None);
+        }
 
-            // Wait for the in progress batch to complete and send a project initialized notification to the client.
-            await ProjectsToLoadAndReload.WaitUntilCurrentBatchCompletesAsync();
-            await ProjectInitializationHandler.SendProjectInitializationCompleteNotificationAsync();
+        var projects = await buildHost.GetProjectsInSolutionAsync(solutionFilePath, CancellationToken.None);
+        foreach (var (path, guid) in projects)
+        {
+            await BeginLoadingProjectAsync(path, guid);
         }
+        await WaitForProjectsToFinishLoadingAsync();
+        await ProjectInitializationHandler.SendProjectInitializationCompleteNotificationAsync();
     }
 
-    public async Task OpenProjectsAsync(ImmutableArray<string> projectFilePaths)
+    public async Task OpenProjectsAsync(ImmutableArray<(string ProjectPath, string? ProjectGuid)> projectFilePaths)
```

Comment:
See comment elsewhere: the place where this gets called the tuple was always had a null ProjectGuid. Could inline that easily enough.